### ISSUE TITLE
Associated data doesn't save on a hasOne relationship

### ISSUE THREAD

**Issue by Robertroid** [cakephp/cakephp#12328]
This is a:

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.6.4
* Platform and Target: Mac OS High Sierra 10.13.3, Apache 2.4, PHP 7.2.6, MariaDB 10.3.7

### What you did
I have entities called _Dibujos_ (Drawings) and _Páginas_ (Pages). Both are linked with _Noticias_ (news) as 1-to-1 (_Dibujos_ and _Páginas_ have hasOne, while _Noticias_ have belongsTo to both of them) through the following conditions:
- For _Páginas_:
`"Noticias.tipo_nodo = 'mangas'"`
- For _Dibujos_:
`"Noticias.tipo_nodo IN ('bocetos','fanart','ilustraciones')"`

### What happened
I wanted to save a _Noticia_ though the data array from _Dibujos_. The save was successful for _Dibujo_ but not for _Noticia_. No errors were thrown.

### What you expected to happen
To save _Dibujo_ and _Noticia_ through associated data with a **hasOne()** relationship

In summary, hasOne() doesn't let me to bulk save associated data. Even if I create an entity _Noticia_ under _Dibujo_ and pass it the data, it doesn't save it. But if I change to hasMany() it works. It seems that the conditions don't play a role in it, as I commented them out and still didn't work.
The saving data that I'm patching to the _Dibujo_ entity is as follows:

    [
	    'id_tipo' => '1',
	    'puntos' => '{"x":246,"y":29,"x2":366,"y2":215,"w":120,"h":186}',
	    'noticia' => [
		    'puntos' => '{"x":203,"y":-5.684341886080802e-14,"x2":331.6170212765958,"y2":325,"w":128.61702127659578,"h":325.00000000000006}',
		    'cabecera' => 'esp',
		    '_translations' => [
			    'en' => [
				    'cabecera' => 'eng'
			    ]
		    ],
		    'tipo_nodo' => 'bocetos'
	    ],
	    'accion' => 'agregar',
	    'imagen' => [
		    'tmp_name' => '/private/var/tmp/phpWBDHcX',
		    'error' => (int) 0,
		    'name' => '14026440_f520.jpg',
		    'type' => 'image/jpeg',
		    'size' => (int) 31790
	    ]
    ]

**Comment by markstory** [comment]
How are you calling patchEntity(), and does the entity you create with patchEntity/newEntity have the `noticia` association populated?

**Comment by Robertroid** [comment]
Ok, I caught something veeery strange:
While I was rolling back all posterior tries that I did after posting this message, I checked that effectively was not saving the new _noticia_ (and it didn't save as expected).

Keep in mind that the relationship is: _Dibujos_ `hasOne` _Noticia_
And this is the code where I was saving (please take special note to the commented line):

    $dibujo = $this->Dibujos->newEntity();
    if ($this->request->is(['post', 'put'])) {
        $data = $this->request->getData();
        $tipoDibujo = $this->Dibujos->Tipo->get($data['id_tipo']);
        $data['noticia']['tipo_nodo'] = $tipoDibujo->codigo; //here it adds "boceto" to "tipo_nodo"
        //$data['noticias'] = $data['noticia']; *take note of this line
        if ($this->request->getData('accion') === 'editar') { //edit action
            $dibujo = $dibujo = $this->Dibujos->find('all')
                ->where(['tipo' => $data['id_tipo']])
                ->order('id desc')
                ->limit('1')
                ->page($data['reemplazo'])
                ->first();
        }
        $dibujo = $this->Dibujos->patchEntity($dibujo, $data);

        if ($resultado = $this->Dibujos->save($dibujo)) {
            $this->Flash->success('El dibujo ha sido guardado exitosamente');
            $this->redirect(['controller' => 'dashboard', 'action' => 'index']);
        } else {
            $this->Flash->error('No se pudo guardar el dibujo. Por favor intente nuevamente');
        }
    }

(By the way, the _Dibujo_ relationship is as follows)

     $this->hasOne('Noticia', [
          'className' => 'Noticias',
          'foreignKey' => 'id_nodo'
     ])->setConditions([
          "Noticias.tipo_nodo IN ('fanart','bocetos','ilustraciones')"
     ]);

Now, if I change the relationship to `hasOne Noticias`, like in plural, and uncomment the previous line, then an entity will be created **on _Noticia_**

Here's what a nice debug() tells me:

### On the first case (with reltionship as 'Noticia')
    object(Cake\ORM\Entity) {
        'id_tipo' => (int) 2,
        'puntos' => '{"x":188,"y":9,"x2":308,"y2":195,"w":120,"h":186}',
        'noticia' => [
            'puntos' => '{"x":160,"y":-5.684341886080802e-14,"x2":288.6170212765958,"y2":325,"w":128.61702127659578,"h":325.00000000000006}',
            'cabecera' => 'asc',
            '_translations' => [
                'en' => [
                    'cabecera' => 'casdfaf'
                ]
            ],
            'tipo_nodo' => 'ilustraciones'
        ],
        'accion' => 'agregar',
	...

###On the second case (with relationship as 'Noticias')

    object(Cake\ORM\Entity) {
        'id_tipo' => (int) 2,
        'puntos' => '{"x":199,"y":57,"x2":319,"y2":243,"w":120,"h":186}',
        'noticia' => object(App\Model\Entity\Noticia) {
            'puntos' => '{"x":154,"y":-5.684341886080802e-14,"x2":282.6170212765958,"y2":325,"w":128.61702127659578,"h":325.00000000000006}',
            'cabecera' => 'awdawdaw',
            '_translations' => [
                'en' => object(App\Model\Entity\Noticia) {
                    'cabecera' => 'sdasd',
                    ...
                }
            ...
        },
        'accion' => 'agregar',
        'imagen' => [
            ...
        ],
        'noticias' => [
            'puntos' => '{"x":154,"y":-5.684341886080802e-14,"x2":282.6170212765958,"y2":325,"w":128.61702127659578,"h":325.00000000000006}',
            'cabecera' => 'awdawdaw',
            '_translations' => [
                'en' => [
                    'cabecera' => 'sdasd'
                ]
            ],
            'tipo_nodo' => 'ilustraciones'
        ],
	...

So, in summary, if I copy the data on the plural key, it will make the entity on the singular key.

**Comment by markstory** [comment]
You might need to define `propertyKey` in your association to ensure that the marshaller/associations set the right key. In order to get the ORM to save an Association you'll need to get the marshaller to convert it into an entity.

I'm going to close this issue as it seems like this is an inflection issue that is related to the specific association names you're using.
